vpr/src/pack/cluster_legalizer.cpp

```diff
@@ -997,11 +997,70 @@ static void update_molecule_chain_info(t_pack_molecule* chain_molecule, const t_
     VTR_ASSERT(false);
 }
 
+/**
+ * @brief Revalidate the molecules associated with this pb. The mol_validated
```

Comment:
If you know anything more about what this routine does (how does it record that molecules are valid? what does it do if mol_validated is true?) it would be good to expand this comment.  If you don't know (you're just moving the code), then we can leave the comment as is.

Reply:
I actually knew very little about this routine. In all honesty it made no sense. I added the "mol_validated" flag as a hot fix to keep the behaviour exactly the same. I agree that it was a rough fix. I decided to dig into this method in detail and found that it was completely not necessary anymore (given my recent changes). I greatly simplified the method and turned it into a "reset_molecule_info" method, which does basically the same thing.